test/javascript-wrapper-mocha.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * File: javascript-wrapper.js
+ * Unit tests for the APIs defined in src/javascript-wrapper.js
+ */
+
+var chai = require('chai');
+
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+
+var assert = chai.asert;
+var expect = chai.expect;
+var should = chai.should();
+
+var _ = require('underscore');
+
+var jswrapper = require('../src/javascript-wrapper');
+var commonTest = require('./common-test');
+
+describe('javascript-wrapper', function() {
+
+    describe('#module.export', function() {
+        it('should exist as a function', function() {
+            jswrapper.should.exist;
+            jswrapper.should.be.a('function');
+        });
+
+        it("should use default port & default updater if no node definition", function() {
+            return Promise.resolve()
+            .then(jswrapper).then(commonTest.createComponent).then(function(component){
+                return _.keys(component.inPorts);
+            }).should.eventually.contain('input');
+        });
+
+        it("should use default port & default updater if empty node definition", function() {
+            return Promise.resolve({})
+            .then(jswrapper).then(commonTest.createComponent).then(function(component){
+                return _.keys(component.inPorts);
+            }).should.eventually.contain('input');
+        });
+
+        it("should use default port & default updater if empty inPorts", function() {
+            return Promise.resolve({inPorts:{}})
+            .then(jswrapper).then(commonTest.createComponent).then(function(component){
+                return _.keys(component.inPorts);
+            }).should.eventually.contain('input');
+        });
+
+        it("should accept array of inPort names", function() {
+            return Promise.resolve({
+                inPorts:['testport']
+            }).then(jswrapper).then(commonTest.createComponent).then(function(component){
+                return _.keys(component.inPorts).sort();
+            }).should.eventually.become(['input', 'testport']);
+        });
+
+        it("should accept object with one inPort definition", function() {
+            return Promise.resolve({
+                inPorts: { 
+                    input: { datatype: 'string', 
```

Comment:
Suggestion: Use a different port name, so that it is distinguishable from the default case.